korman/exporter/etlight.py

```diff
@@ -384,25 +386,25 @@ def _prep_for_lightmap(self, bo, toggle):
             uv_textures.active = uvtex
 
             # if the artist hid any UVs, they will not be baked to... fix this now
-            bpy.ops.object.mode_set(mode="EDIT")
-            bpy.ops.uv.reveal()
-            bpy.ops.object.mode_set(mode="OBJECT")
+            with self._set_mode("EDIT"):
+                bpy.ops.uv.reveal()
             self._associate_image_with_uvtex(uv_textures.active, im)
-            bpy.ops.object.mode_set(mode="EDIT")
 
-            # prep the uvtex for lightmapping
-            bpy.ops.mesh.select_all(action="SELECT")
-            bpy.ops.uv.average_islands_scale()
-            bpy.ops.uv.pack_islands()
+            # Meshes with modifiers need to have islands packed to prevent generated vertices
+            # from sharing UVs. Sigh.
+            if self.is_collapsed(bo):
+                self._report.warn("'{}': packing islands in UV Texture '{}' due to modifier collapse", bo.name, uv_base.name)
+                with self._set_mode("EDIT"):
+                    bpy.ops.mesh.select_all(action="SELECT")
+                    bpy.ops.uv.pack_islands()
```

Comment:
```suggestion
                    bpy.ops.uv.pack_islands(margin=0.01)
```
The default of 0.001 packs islands a bit too tight IMHO. 0.01 should help avoid pixels leaking between islands, especially when dealing with many islands or small lightmap sizes.